### ISSUE TITLE
bug: cannot change api url

### ISSUE THREAD

**Issue by ego-lay-atman-bay** [ReVanced/revanced-manager#1434]
### Bug description

I went to go change the api url from `releases.revanced.app` to `api.revanced.app`, but I found out I actually can't. ReVanced manager does not update the api url no matter what I try.

### Version of ReVanced Manager and version & name of application you tried to patch

ReVanced Manager: 1.14.1

### Installation type

Non-root

### Device logs

```shell
--------- beginning of main
10-26 01:02:38.977 24576 24576 I ImeTracker: app.revanced.manager.flutter:3f3862f3: onRequestShow at ORIGIN_CLIENT_SHOW_SOFT_INPUT reason SHOW_SOFT_INPUT
10-26 01:02:38.995 24576 24576 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.FlutterView{5e1fea8 VFED..... .F....ID 0,0-1080,2400 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
10-26 01:02:39.007 24576 24576 I AssistStructure: Flattened final assist data: 528 bytes, containing 1 windows, 3 views
10-26 01:02:39.027 24576 24576 D InsetsController: show(ime(), fromIme=true)
10-26 01:02:39.348 24576 24576 I ImeTracker: app.revanced.manager.flutter:3f3862f3: onShown
10-26 01:02:40.076 24576 24576 I ImeTracker: app.revanced.manager.flutter:4b75ce30: onRequestShow at ORIGIN_CLIENT_SHOW_SOFT_INPUT reason SHOW_SOFT_INPUT
10-26 01:02:40.076 24576 24576 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.FlutterView{5e1fea8 VFED..... .F...... 0,0-1080,2400 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
10-26 01:02:40.078 24576 24576 D InsetsController: show(ime(), fromIme=true)
10-26 01:02:40.078 24576 24576 I ImeTracker: app.revanced.manager.flutter:4b75ce30: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-26 01:02:40.420 24576 24576 I ImeTracker: app.revanced.manager.flutter:ce8f82d2: onRequestShow at ORIGIN_CLIENT_SHOW_SOFT_INPUT reason SHOW_SOFT_INPUT
10-26 01:02:40.421 24576 24576 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.FlutterView{5e1fea8 VFED..... .F...... 0,0-1080,2400 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
10-26 01:02:40.427 24576 24576 D InsetsController: show(ime(), fromIme=true)
10-26 01:02:40.427 24576 24576 I ImeTracker: app.revanced.manager.flutter:ce8f82d2: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-26 01:02:40.470 24576 24576 I ImeTracker: app.revanced.manager.flutter:48371b70: onRequestShow at ORIGIN_CLIENT_SHOW_SOFT_INPUT reason SHOW_SOFT_INPUT
10-26 01:02:40.470 24576 24576 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.FlutterView{5e1fea8 VFED..... .F...... 0,0-1080,2400 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
10-26 01:02:40.475 24576 24576 D InsetsController: show(ime(), fromIme=true)
10-26 01:02:40.475 24576 24576 I ImeTracker: app.revanced.manager.flutter:48371b70: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-26 01:02:40.520 24576 24576 I ImeTracker: app.revanced.manager.flutter:17fd7835: onRequestShow at ORIGIN_CLIENT_SHOW_SOFT_INPUT reason SHOW_SOFT_INPUT
10-26 01:02:40.520 24576 24576 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.FlutterView{5e1fea8 VFED..... .F...... 0,0-1080,2400 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
10-26 01:02:40.525 24576 24576 D InsetsController: show(ime(), fromIme=true)
10-26 01:02:40.525 24576 24576 I ImeTracker: app.revanced.manager.flutter:17fd7835: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-26 01:02:40.571 24576 24576 I ImeTracker: app.revanced.manager.flutter:76a658a1: onRequestShow at ORIGIN_CLIENT_SHOW_SOFT_INPUT reason SHOW_SOFT_INPUT
10-26 01:02:40.571 24576 24576 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.FlutterView{5e1fea8 VFED..... .F...... 0,0-1080,2400 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
10-26 01:02:40.576 24576 24576 D InsetsController: show(ime(), fromIme=true)
10-26 01:02:40.576 24576 24576 I ImeTracker: app.revanced.manager.flutter:76a658a1: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-26 01:02:40.621 24576 24576 I ImeTracker: app.revanced.manager.flutter:4e865e91: onRequestShow at ORIGIN_CLIENT_SHOW_SOFT_INPUT reason SHOW_SOFT_INPUT
10-26 01:02:40.621 24576 24576 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.FlutterView{5e1fea8 VFED..... .F...... 0,0-1080,2400 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
10-26 01:02:40.626 24576 24576 D InsetsController: show(ime(), fromIme=true)
10-26 01:02:40.626 24576 24576 I ImeTracker: app.revanced.manager.flutter:4e865e91: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-26 01:02:40.674 24576 24576 I ImeTracker: app.revanced.manager.flutter:a2f0bba8: onRequestShow at ORIGIN_CLIENT_SHOW_SOFT_INPUT reason SHOW_SOFT_INPUT
10-26 01:02:40.674 24576 24576 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.FlutterView{5e1fea8 VFED..... .F...... 0,0-1080,2400 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
10-26 01:02:40.682 24576 24576 D InsetsController: show(ime(), fromIme=true)
10-26 01:02:40.682 24576 24576 I ImeTracker: app.revanced.manager.flutter:a2f0bba8: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-26 01:02:40.725 24576 24576 I ImeTracker: app.revanced.manager.flutter:87ea4c8b: onRequestShow at ORIGIN_CLIENT_SHOW_SOFT_INPUT reason SHOW_SOFT_INPUT
10-26 01:02:40.725 24576 24576 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.FlutterView{5e1fea8 VFED..... .F...... 0,0-1080,2400 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
10-26 01:02:40.733 24576 24576 D InsetsController: show(ime(), fromIme=true)
10-26 01:02:40.733 24576 24576 I ImeTracker: app.revanced.manager.flutter:87ea4c8b: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-26 01:02:41.415 24576 24576 I ImeTracker: app.revanced.manager.flutter:4d941757: onRequestShow at ORIGIN_CLIENT_SHOW_SOFT_INPUT reason SHOW_SOFT_INPUT
10-26 01:02:41.415 24576 24576 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.FlutterView{5e1fea8 VFED..... .F...... 0,0-1080,2400 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
10-26 01:02:41.428 24576 24576 D InsetsController: show(ime(), fromIme=true)
10-26 01:02:41.428 24576 24576 I ImeTracker: app.revanced.manager.flutter:4d941757: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-26 01:02:41.785 24576 24576 I ImeTracker: app.revanced.manager.flutter:31ed21f4: onRequestShow at ORIGIN_CLIENT_SHOW_SOFT_INPUT reason SHOW_SOFT_INPUT
10-26 01:02:41.785 24576 24576 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.FlutterView{5e1fea8 VFED..... .F...... 0,0-1080,2400 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
10-26 01:02:41.788 24576 24576 D InsetsController: show(ime(), fromIme=true)
10-26 01:02:41.788 24576 24576 I ImeTracker: app.revanced.manager.flutter:31ed21f4: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-26 01:02:42.191 24576 24576 I ImeTracker: app.revanced.manager.flutter:c10a2285: onRequestShow at ORIGIN_CLIENT_SHOW_SOFT_INPUT reason SHOW_SOFT_INPUT
10-26 01:02:42.191 24576 24576 D InputMethodManager: showSoftInput() view=io.flutter.embedding.android.FlutterView{5e1fea8 VFED..... .F...... 0,0-1080,2400 #1 aid=1073741824} flags=0 reason=SHOW_SOFT_INPUT
10-26 01:02:42.195 24576 24576 D InsetsController: show(ime(), fromIme=true)
10-26 01:02:42.195 24576 24576 I ImeTracker: app.revanced.manager.flutter:c10a2285: onCancelled at PHASE_CLIENT_APPLY_ANIMATION
10-26 01:02:43.028 24576 24635 E flutter : [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: LateInitializationError: Field '_dio@975503907' has already been initialized.
10-26 01:02:43.028 24576 24635 E flutter : #0      LateError._throwFieldAlreadyInitialized (dart:_internal-patch/internal_patch.dart:192)
10-26 01:02:43.028 24576 24635 E flutter : #1      RevancedAPI._dio= (package:revanced_manager/services/revanced_api.dart:16)
10-26 01:02:43.028 24576 24635 E flutter : #2      RevancedAPI.initialize (package:revanced_manager/services/revanced_api.dart:22)
10-26 01:02:43.028 24576 24635 E flutter : #3      ManagerAPI.setApiUrl (package:revanced_manager/services/manager_api.dart:82)
10-26 01:02:43.028 24576 24635 E flutter : #4      SManageApiUrl.showApiUrlDialog.<anonymous closure>.<anonymous closure> (package:revanced_manager/ui/views/settings/settingsFragment/settings_manage_api_url.dart:69)
10-26 01:02:43.028 24576 24635 E flutter : #5      _InkResponseState.handleTap (package:flutter/src/material/ink_well.dart:1154)
10-26 01:02:43.028 24576 24635 E flutter : #6      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:275)
10-26 01:02:43.028 24576 24635 E flutter : #7      TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:654)
10-26 01:02:43.028 24576 24635 E flutter : #8      BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:311)
10-26 01:02:43.028 24576 24635 E flutter : #9      BaseTapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:244)
10-26 01:02:43.028 24576 24635 E flutter : #10     PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:630)
10-26 01:02:43.028 24576 24635 E flutter : #11     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:98)
10-26 01:02:43.028 24576 24635 E flutter : #12     PointerRouter._dispatchEventToRoutes.<anonymous closure> (package:flutter/src/gestures/pointer_router.dart:143)
10-26 01:02:43.028 24576 24635 E flutter : #13     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:625)
10-26 01:02:43.028 24576 24635 E flutter : #14     PointerRouter._dispatchEventToRoutes (package:flutter/src/gestures/pointer_router.dart:141)
10-26 01:02:43.028 24576 24635 E flutter : #15     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:127)
10-26 01:02:43.028 24576 24635 E flutter : #16     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:488)
10-26 01:02:43.028 24576 24635 E flutter : #17     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:468)
10-26 01:02:43.028 24576 24635 E flutter : #18     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:333)
10-26 01:02:43.028 24576 24635 E flutter : #19     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:413)
10-26 01:02:43.028 24576 24635 E flutter : #20     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:376)
10-26 01:02:43.028 24576 24635 E flutter : #21     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:323)
10-26 01:02:43.028 24576 24635 E flutter : #22     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:292)
10-26 01:02:43.028 24576 24635 E flutter : #23     _invoke1 (dart:ui/hooks.dart:186)
10-26 01:02:43.028 24576 24635 E flutter : #24     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:4
```

### Acknowledgements

- [X] This request is not a duplicate of an existing issue.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] The issue is solely related to the ReVanced Manager

**Comment by validcube** [comment]
👋 I would recommend that you try turning on Airplane mode (or disable your internet connection), and open ReVanced Manager again.

If you have turned on Airplane mode and ReVanced Manager still crashes, force exit the application and start again.

**Comment by ego-lay-atman-bay** [comment]
It's not crashing, it's just not updating the url. Like, when I input `api.revanced.app`, hit ok, tap to change the api url again, it's still `releases.revanced.app`. Even when I restart the app it doesn't change it.

**Comment by validcube** [comment]
> It's not crashing, it's just not updating the url. Like, when I input `api.revanced.app`, hit ok, tap to change the api url again, and it's still `releases.revanced.app`. Even when I restart the app it doesn't change it.

Ah whoops, have you pressed the reset button and restart? (I assume you have, but just to sure.)

**Comment by ego-lay-atman-bay** [comment]
> have you pressed the reset button and restart?

Yup. I have even reinstalled the app just in case it was something to do with updating the app, but it still wouldn't change the api (and now I can't get any patches because I can't fix the api url).

**Comment by Ushie** [comment]
What about clear data? some ROMs keep some app data regardless of reinstall

**Comment by ego-lay-atman-bay** [comment]
I should probably have mentioned before, I'm on google pixel 6a and android 14. I'm pretty sure I'm just on the stock android (or the google version).

**Comment by Ushie** [comment]
Releasing v1.14.2 with a fix, thanks to @TheAabedKhan

**Comment by Ushie** [comment]
Please test on v1.14.2 and close this issue if it's resolved (misclick)

**Comment by ego-lay-atman-bay** [comment]
The issue is fixed in v1.14.2
